File: ee25btech11019/matgeo/2.7.31/codes/4.c
#include <stdio.h>
#include <math.h>

// Function to compute area of triangle from 3 points
double triangle_area(double x1, double y1, double x2, double y2, double x3, double y3) {
    double det = x1*(y2 - y3) - y1*(x2 - x3) + (x2*y3 - y2*x3);
    return fabs(det) / 2.0;
}

int main() {
    double A[2] = {2, 5};
    double B[2] = {4, 7};
    double C[2] = {6, 2};

    double area = triangle_area(A[0], A[1], B[0], B[1], C[0], C[1]);
    printf("Area of triangle ABC = %.2f\n", area);
    return 0;
}

File: ee25btech11019/matgeo/2.7.31/codes/4.py
import numpy as np
import matplotlib.pyplot as plt

# Points
A = np.array([2, 5])
B = np.array([4, 7])
C = np.array([6, 2])

# Area using determinant method
matrix = np.array([
    [A[0], A[1], 1],
    [B[0], B[1], 1],
    [C[0], C[1], 1]
])
area = 0.5 * abs(np.linalg.det(matrix))
print(f"Area of triangle ABC = {area:.2f}")

# Plot triangle
x_coords = [A[0], B[0], C[0], A[0]]
y_coords = [A[1], B[1], C[1], A[1]]

plt.plot(x_coords, y_coords, 'b-', label='Triangle ABC')
plt.scatter([A[0], B[0], C[0]], [A[1], B[1], C[1]], color='red')

# Annotate points
labels = ['A(2,5)', 'B(4,7)', 'C(6,2)']
for (x, y), label in zip([A, B, C], labels):
    plt.annotate(label, (x, y), textcoords="offset points", xytext=(5,5), ha='center')

plt.title("Triangle ABC")
plt.xlabel("x")
plt.ylabel("y")
plt.grid(True)
plt.axis('equal')
plt.legend()
plt.savefig("triangle.png")
plt.show()
